Add Page Visibility API iframe Site Isolation Test

stwrt opened the pull request:
#### fc84bdd750e19392c947e21685b02f64b36e8e88
<pre>
Add Page Visibility API iframe Site Isolation Test
<a href="https://bugs.webkit.org/show_bug.cgi?id=287569">https://bugs.webkit.org/show_bug.cgi?id=287569</a>
<a href="https://rdar.apple.com/problem/144710517">rdar://problem/144710517</a>

Reviewed by Alex Christensen.

Add test case to ensure that the page visibility api test works when in a remote iframe under site isolation.

On iOS, the `testRunner.setPageVisibility(&quot;hidden&quot;);` api is not working w/ or w/o site isolation, which
is causing the test to fail.

* LayoutTests/http/tests/site-isolation/page-visibility-onvisibilitychange-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/page-visibility-onvisibilitychange-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/resources/page-visibility-iframe-send-onvisibility-change-back.html: Added.
* LayoutTests/platform/ios/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/291497@main">https://commits.webkit.org/291497@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78dcbad526d4b8b6ce8962721b37b2cb1da62eab

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93077 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/12628 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/2333 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98076 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/43603 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/12909 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21084 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/71167 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28586 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96079 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/9737 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84227 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51495 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9430 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/1861 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/42916 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79721 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/1839 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100103 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20129 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/14768 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/80194 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/20381 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80125 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79495 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19738 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24039 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1357 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13222 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20113 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/19800 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23260 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/21541 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on LayoutTests/fast/events/page-visibility-onvisibilitychange-in-iframe.html:
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html>
+<body onload="startTest()">
+<iframe id="iframeTest" src="resources/page-visibility-iframe-send-onvisibility-change-back.html"></iframe>
```
This is not a cross-site iframe.  We need something like LayoutTests/http/tests/site-isolation/basic-iframe.html that uses localhost for the iframe and implicitly 127.0.0.1 for the main frame.

stwrt replied:
Fixed

stwrt committed:
Add Page Visibility API iframe Site Isolation Test
https://bugs.webkit.org/show_bug.cgi?id=287569
rdar://problem/144710517

Reviewed by Alex Christensen.

Add test case to ensure that the page visibility api test works when in a remote iframe under site isolation.

On iOS, the `testRunner.setPageVisibility("hidden");` api is not working w/ or w/o site isolation, which
is causing the test to fail.

* LayoutTests/http/tests/site-isolation/page-visibility-onvisibilitychange-in-iframe-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/page-visibility-onvisibilitychange-in-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/resources/page-visibility-iframe-send-onvisibility-change-back.html: Added.
* LayoutTests/platform/ios/TestExpectations:

Canonical link: https://commits.webkit.org/291497@main
